pandas/io/formats/style.py

```diff
@@ -1569,34 +1569,67 @@ def set_sticky(
                 # but <th> will exist in <thead>: conflict with initial element
                 styles = [
                     {
-                        "selector": "tr th:first-child",
+                        "selector": "thead tr th:nth-child(1)",
+                        "props": props + "left:0px; z-index:3 !important;",
+                    },
+                    {
+                        "selector": "tbody tr th:nth-child(1)",
                         "props": props + "left:0px; z-index:1;",
-                    }
+                    },
                 ]
 
-            return self.set_table_styles(styles, overwrite=False)
-
         else:
+            # handle the MultiIndex case
             range_idx = list(range(obj.nlevels))
+            levels = sorted(levels) if levels else range_idx
 
-        levels = sorted(levels) if levels else range_idx
-        for i, level in enumerate(levels):
-            self.set_table_styles(
-                [
-                    {
-                        "selector": f"{tag} th.level{level}",
-                        "props": f"position: sticky; "
-                        f"{pos}: {i * pixel_size}px; "
-                        f"{f'height: {pixel_size}px; ' if axis == 1 else ''}"
-                        f"{f'min-width: {pixel_size}px; ' if axis == 0 else ''}"
-                        f"{f'max-width: {pixel_size}px; ' if axis == 0 else ''}"
-                        f"background-color: white;",
-                    }
-                ],
-                overwrite=False,
-            )
+            if axis == 1:
```

Comment:
can this be in the axis=1 clause above?

Reply:
there are dual conditionals:
```
if MultiIndex:
    if axis==1: ...
    else: ...
else:
    if axis==1: ...
    else:...
```
so it could be structured differently but I think the code is a bit clearer as is